cm-gateway/cm-config/terragrunt.hcl

```diff
@@ -0,0 +1,37 @@
+dependency "vault-install" {
+  config_path = "../cm-install"
+
+  skip_outputs = "true"
+}
+
+terraform {
+  extra_arguments "common_vars" {
+    commands = [
+      "apply",
+      "plan",
+      "import",
+      "push",
+      "refresh",
+      "destroy"
+    ]
+
+    arguments = [
+      "-var-file=${get_parent_terragrunt_dir()}/../common.tfvars"
+    ]
+  }
+
+  extra_arguments "silence-warnings" {
+    commands = [
+      "apply",
+      "plan",
+      "import",
+      "push",
+      "refresh",
+      "destroy"
+    ]
+
+    arguments = [
+      "-compact-warnings"
+    ]
+  }
+}
```

Comment:
> do you think we need to do this?

My main consideration is that these `Value for undeclared variable` warnings will take up a lot of screen space. With the increasement of `common variable`, the more space will be occupied.
```
......
Terraform has compared your real infrastructure against your configuration
and found no differences, so no changes are needed.
╷
│ Warning: Value for undeclared variable
│
│ The root module does not declare a variable named "cm_version" but a value
│ was found in file
│ "/root/lonelyCZ/testing-addons/cm-gateway/gateway-api-install/../common.tfvars".
│ If you meant to use this value, add a "variable" block to the
│ configuration.
│
│ To silence these warnings, use TF_VAR_... environment variables to provide
│ certain "global" settings to all configurations in your organization. To
│ reduce the verbosity of these warnings, use the -compact-warnings option.
╵
╷
│ Warning: Value for undeclared variable
│
│ The root module does not declare a variable named "demo_namespace" but a
│ value was found in file
│ "/root/lonelyCZ/testing-addons/cm-gateway/gateway-api-install/../common.tfvars".
│ If you meant to use this value, add a "variable" block to the
│ configuration.
│
│ To silence these warnings, use TF_VAR_... environment variables to provide
│ certain "global" settings to all configurations in your organization. To
│ reduce the verbosity of these warnings, use the -compact-warnings option.
╵
╷
│ Warning: Values for undeclared variables
│
│ In addition to the other similar warnings shown, 2 other variable(s)
│ defined without being declared.
......
```

>  I think it might be preferable to see the warnings so as to not miss any useful information if things fail

If some things fail, it can unfold the warns automatically.

```
......
null_resource.wait_for_contour (local-exec): Error from server (NotFound): namespaces "projectcontour" not found
╷
│ Warning: Value for undeclared variable
│
│ The root module does not declare a variable named "demo_namespace" but a
│ value was found in file
│ "/root/lonelyCZ/testing-addons/cm-gateway/gateway-controller-install/../common.tfvars".
│ If you meant to use this value, add a "variable" block to the
│ configuration.
│
│ To silence these warnings, use TF_VAR_... environment variables to provide
│ certain "global" settings to all configurations in your organization. To
│ reduce the verbosity of these warnings, use the -compact-warnings option.
╵
......
```

> If there are any warnings that always appear we should probably look at fixing the underlying issue rather than suppressing the warnings- if possible.

Yes, I think so. But due to https://github.com/cert-manager/testing-addons/pull/14#issuecomment-1178827500, I have not found a good way to balance them.

Reply:
I had another look at this and I see that the flag actually only makes the warning output more compact as opposed to not showing the warnings at all https://github.com/hashicorp/terraform/pull/23632. In that case, I think this makes sense 👍🏼 

Could we please rename this block? 'silence-warnings' suggests that the warnings are not shown. Maybe it could be 'compact-warnings' or anything else that suggests that this makes the warnings output more compact.